docs/source/links/state.md

```diff
@@ -414,30 +447,32 @@ write any data. Let's look at an example where we add a todo to a list:
 let nextTodoId = 0;
 
 const todos = {
-  Query: {
-    todos: () => [],
+  defaults: {
+    todos: [],
   },
-  Mutation: {
-    addTodo: (_, { text }, { cache }) => {
-      const query = gql`
-        query GetTodos {
-          todos @client {
-            id
-            text
-            completed
+  resolvers: {
+    Mutation: {
+      addTodo: (_, { text }, { cache }) => {
+        const query = gql`
+          query GetTodos {
+            todos @client {
+              id
+              text
+              completed
+            }
           }
-        }
-      `;
-      const previous = cache.readQuery({ query });
-      const data = {
-        todos: previous.todos.concat([
-          { id: nextTodoId++, text, completed: false, __typename: 'TodoItem' },
-        ]),
-      };
-
-      // you can also do cache.writeData({ data }) here if you prefer
-      cache.writeQuery({ query, data });
-      return data;
+        `;
+
+        const previous = cache.readQuery({ query });
+        const data = {
+          todos: previous.todos.concat([
+            { id: nextTodoId++, text, completed: false, __typename: 'TodoItem' },
+          ]),
+        };
+
+        // you can also do cache.writeData({ data }) here if you prefer
+        cache.writeQuery({ query, data });
```

Comment:
shouldn't we return data here so the user has access to the new todo's id?

Reply:
Oh snap you're right. maybe let's return just the ID and not all the data?